subtests/docker_cli/systemd/systemd.py

```diff
@@ -65,22 +58,32 @@ def sysd_action(action=None, sysd_unit=None):
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def write_unitfile(self, var_to_edit):
+        with open(self.sub_stuff['unit_src'], "r") as srcfile:
+            lines = srcfile.readlines()
+        with open(self.sub_stuff['unit_dest'], "w") as tmpfile:
+            for line in lines:
+                for unitf_var_name, changeto in var_to_edit.items():
+                    if unitf_var_name in line:
+                        str_to_change = \
+                            line.partition(unitf_var_name)[2].strip()
+                        line = line.replace(str_to_change, changeto)
```

Comment:
One more clue.  Lines w/o tokens will throw exceptions, which can be made useful like this:

```python
try:
     replacement_line = ...super secret sauce...
except KeyError:
    replacement_line = original_line
...
unitfile.write(replacement_line)
```
```